src/index.ts

```diff
@@ -34,6 +34,13 @@ export type FieldState<Value, ErrorMessage = string> = {
   error: ErrorMessage | undefined;
 };
 
+type ListenFieldsFunction<Values extends Record<string, unknown>, ErrorMessage = string> = <
```

Comment:
There's no need to extract the type, you can inline it inside the `Form` type, like other `api` methods